src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -298,13 +298,30 @@ private void RemoveEntry(NodeSearchElement entry)
             var treeStack = new Stack<NodeCategoryViewModel>();
             var nameStack = new Stack<string>(entry.Categories.Reverse());
             var target = libraryRoot;
+            bool isCheckedForClassesCategory = false;
             while (nameStack.Any())
             {
                 var next = nameStack.Pop();
                 var categories = target.SubCategories;
                 var newTarget = categories.FirstOrDefault(c => c.Name == next);
-                if (newTarget == default(NodeCategoryViewModel))
+                if (newTarget == null)
                 {
+                    // Latest item in categories list can be a class.
+                    // All classes are situated in classes container.
+                    // So nameStack should be populated with ClassesDefaultName
+                    // and class name. And checked again.
+                    // For example: "MyAssembly.MyNamespace.ClassCandidate" not found.
+                    // Should be checked for "MyAssembly.MyNamespace.Classes.ClassCandidate"
+                    //
```

Comment:
I'd recommend the following changes:

```csharp
// The last entry in categories list can be a class name. When the desired class 
// cannot be located with "MyAssembly.MyNamespace.ClassCandidate" pattern, try 
// searching with "MyAssembly.MyNamespace.Classes.ClassCandidate" instead. This 
// is because a class always resides under a "ClassesNodeCategoryViewModel" node.
// 
```